var/spack/repos/builtin/packages/hadoop/package.py

```diff
@@ -13,8 +13,10 @@ class Hadoop(Package):
     """
 
     homepage = "https://hadoop.apache.org/"
-    url      = "https://www.apache.org/dist/hadoop/common/hadoop-3.2.1/hadoop-3.2.1.tar.gz"
+    urls     = ["https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2.tar.gz",
+                "https://downloads.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2.tar.gz"]
 
+    version('3.3.2',  sha256='b341587495b12eec0b244b517f21df88eb46ef634dc7dc3e5969455b80ce2ce5')
```

Comment:
I am unable to confirm the sha256 using `spack checksum` for this or any of the other available versions using these urls due to `The read operation timed out` errors.

I *can* `spack checksum` the ones at those sites with the following urls (i.e., when I add `-src` before `.tar.gz`) but get a different checksum:

```suggestion
    urls     = ["https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-src.tar.gz",
                "https://downloads.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-src.tar.gz"]

    version('3.3.2',  sha256='f0140da77581c6cdcc64753187f7563a2ab616a645e086fe78fa4a7b1d45c6de')
```

Note that I *can* still access older, though not necessarily recommended versions if I add `"https://archive.apache.org/dist/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-src.tar.gz"` to the `urls` list.  BUT that also means all of the checksums would need to be modified.

Reply:
Thanks for helping me, @tldahlgren ! Before adding your suggested changes:
```
$ spack checksum hadoop
==> Found 2 versions of hadoop:
  
  3.3.2-site  https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-site.tar.gz
  3.3.2       https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2.tar.gz

==> How many would you like to checksum? (default is 1, q to abort) 2
==> Fetching https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-site.tar.gz
==> Fetching https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2.tar.gz

    version('3.3.2-site', sha256='0305dade655f6e7c59d5cd085781a2a6c85c4b582e3edc86e250656b2b084ee4')
    version('3.3.2',      sha256='b341587495b12eec0b244b517f21df88eb46ef634dc7dc3e5969455b80ce2ce5')
```
After adding your suggested changes:
```
$ spack checksum hadoop
==> Found 1 version of hadoop:
  
  3.3.2  https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-src.tar.gz

==> Fetching https://dlcdn.apache.org/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-src.tar.gz

    version('3.3.2', sha256='f0140da77581c6cdcc64753187f7563a2ab616a645e086fe78fa4a7b1d45c6de')
```
Yes, the checksum would be different if you add `-src`, but I don't get any problems with the urls I added, so then I don't understand why I should add `-src`?
Also, `spack checksum` doesn't find any other versions adding "https://archive.apache.org/dist/hadoop/common/hadoop-3.3.2/hadoop-3.3.2-src.tar.gz" to the list of urls (maybe because of the folder structure in https://archive.apache.org/dist/hadoop/common/). But I guess I could add the tarball url specifically for each version.

Reply:
🤔 I was providing the version number on the command line rather than depending on Spack to find all possible versions (i.e., I used `spack checksum hadoop 3.3.2 3.3.0 3.2.2 3.2.1` etc.).

I'm not sure if the problem has to do with the network I was on.  I also didn't investigate the difference -- just noted size differences -- between `hadoop-3.3.2.tar.gz` and `hadoop-3.3.2-src.tar.gz`.

Reply:
Looking closer into it, the content in the tarballs with `-src` is fundamentally different, and the spack recipe is made for the ones without `src`, so I would vote for preserving the recipe and not use the `-src` ones. Replacing the old outdated url with "https://archive.apache.org/dist/hadoop/common/hadoop-3.3.2/hadoop-3.3.2.tar.gz" is sufficient for the old checksums to still be valid. Your suggestion surely got me on the right track.
Also, I didn't know the logic was different for `spack checksum` if you specify the version number, so thanks a lot for letting me know!

Reply:
Strange.  I still cannot confirm the checksums using `spack checksum hadoop 3.3.2 3.3.0` *or* `spack checksum hadoop`.  In both cases I get `The read operation timed out.`

But, `spack install --fake hadoop@3.3.2` works so I won't stand in the way of this PR getting merged as, perhaps, it's an issue for me.